src/vkscript/parser.cc

```diff
@@ -372,14 +375,19 @@ Result Parser::ProcessVertexDataBlock(const std::string& data) {
             return Result("Invalid vertex data value");
           }
 
-          row.back().AppendValue(std::move(v));
+          buffer_data.push_back(v);
         }
       }
     }
-    node->AddRow(std::move(row));
   }
 
-  node->SetHeaders(std::move(headers));
+  auto node = MakeUnique<VertexDataNode>();
+  for (size_t i = 0; i < headers.size(); ++i) {
+    auto buffer = MakeUnique<Buffer>(BufferType::kVertex);
+    buffer->SetData(std::move(values[i]));
```

Comment:
The Buffer's DatumType  and Size attributes are not set.  Is that important?

Reply:
I think Buffer needs comments to describe its own internal data,on the data members and accessors.  The particularly odd one is the datum_type_ which should be described as only valid for an index buffer.

Reply:
Added a couple comments. I added the invalid DatumType comment in the VertexData node class as that's where it's invalid. Buffer is also used in the Amber code where the DatumType is valid. It's just with VertexData that it goes wonky due to things like form D24_UINT_8_SINT or D32_SFLOAT_8_UINT not really being representable.....